docs: Add effect v4 support (#17472)
getsentry/sentry-docs: docs/platforms/javascript/guides/effect/index.mdx

This guide will show you how to integrate Sentry into your [Effect](https://effect.website/) project using the `@sentry/effect` SDK.

> `@sentry/effect` supports **Effect v3** and **Effect v4**. The integration detects the installed Effect version at runtime, but the `Tracer` and `Logger` layer APIs differ between major versions, so use the **Effect v3** or **Effect v4** snippets below for your app.

Reviewer: ```suggestion
`@sentry/effect` supports **Effect v3** and **Effect v4**. The integration automatically detects which version you have installed, but the `Tracer` and `Logger` layer APIs differ between major versions. Make sure to follow the correct code snippets for your version below.
```

## Step 2: Configure

> The SDK provides an `effectLayer` that initializes Sentry. You can compose it with additional Effect layers to enable tracing, logging, and metrics. The `effectLayer`, `SentryEffectTracer`, `SentryEffectLogger`, and `SentryEffectMetricsLayer` exports are the same for Effect v3 and v4; only the layer composition around tracing and logging changes.

Reviewer: ```suggestion
The SDK provides an `effectLayer` that initializes Sentry. You can compose it with additional Effect layers to enable tracing, logging, and metrics. The `effectLayer`, `SentryEffectTracer`, `SentryEffectLogger`, and `SentryEffectMetricsLayer` exports are the same for Effect v3 and v4. The only difference between versions is how you compose layers for tracing and logging, which is covered in the snippets below.
```

### Effect v3

<AvailableSince version="10.44.0" />

> On **Node**, provide `SentryLive` before launching your HTTP layer (for example with `NodeRuntime`). In the **browser**, provide `SentryLive` to your app layer. Use `Layer.setTracer` and `Logger.replace` for tracing and logs.

Reviewer: Does tracing and logs affect both Node and browser? Then I would use bullet points for Node and browser and put the sentence about logs and traces below.

```suggestion
- Node: Provide `SentryLive` before launching your HTTP layer (for example, with `NodeRuntime`).
- Browser: Provide `SentryLive` to your app layer.

In both cases, use `Layer.setTracer` to configure tracing and `Logger.replace` to configure logging.
```

```typescript
import { Effect } from "effect";
import * as Sentry from "@sentry/effect";

const program = Effect.gen(function* () {

>   yield* Effect.fail(new Error("Sentry Test Error"));

Author: Both are fine. But good suggestion
